tummensabot/meals.py

```diff
@@ -104,9 +122,36 @@ def is_closed(self):
 
 
 class MenuManager:
+    """Responsible for retrieving menus. Caches up to 20 entries for up to 1 hour."""
+
+    def __init__(self):
+        self.cache = ExpiringDict(max_len=20, max_age_seconds=60 * 60)
+
+    def clear_cache(self):
+        self.cache.clear()
+
+    def get_menu(self, mensa_id: int):
+        initial_day = self.get_day()
+        # cache is indexed by mensa and initial day
+        cache_key = (mensa_id, initial_day.isoformat())
+
+        menu, age = self.cache.get(cache_key, default=None, with_age=True)
+        if menu:
+            # cache hit
+            print(f"Using cached menu, age: {age:.0f} seconds")
```

Comment:
Can you use the logging package with logging.debug etc for all print statements?

Reply:
Done